test/checker/test_native_gate_checker.py

```diff
@@ -0,0 +1,55 @@
+# Tests for native gate checker pass
+
+import math
+
+import pytest
+
+from opensquirrel import CircuitBuilder
+from opensquirrel.circuit import Circuit
+from opensquirrel.ir import Float
+from opensquirrel.passes.checker.native_gate_checker import NativeGateChecker
+
+
+@pytest.fixture(name="checker")
+def checker_fixture() -> NativeGateChecker:
+    native_gate_set = ["H", "Z", "Y", "Rx", "CNOT"]
+    return NativeGateChecker(native_gate_set)
```

Comment:
I've updated the example native gate set with a bit more realistic native gates (`H` and `CNOT` are almost never native gates).

```suggestion
@pytest.fixture(name="validator")
def validator_fixture() -> NativeGateValidator:
    native_gate_set = ["I", "X90", "mX90", "Y90", "mY90", "Rz", "CZ"]
    return NativeGateValidator(native_gate_set)
```

Note that the tests below need to be adjusted according to the new native gate set.